src/scancode/cli.py

```diff
@@ -185,6 +186,11 @@ def validate_formats(ctx, param, value):
         raise click.BadParameter('Invalid template file: "%(value)s" does not exists or is not readable.' % locals())
     return value
 
+def ignore_files(ctx, param, value):
+    for file in value.split(','):
+        user_ignored.update({
```

Comment:
I do not think so , but I am not exactly what you mean. Globals are bad in all cases, callback to update a global is equally bad if not worse.

ignores are a plain dict, just use plain dict  and update it in the `scancode()` main function based on user options.